src/Pretzel.Logic/Commands/PretzelBaseCommandParameters.cs

```diff
@@ -0,0 +1,83 @@
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.IO.Abstractions;
+using System.Linq;
+using Pretzel.Logic.Extensions;
+using Pretzel.Logic.Templating;
+using Pretzel.Logic.Templating.Context;
+
+namespace Pretzel.Logic.Commands
+{
+    public abstract class PretzelBaseCommandParameters : BaseParameters, ISourcePathProvider
+    {
+        protected readonly IFileSystem fileSystem;
+
+        protected PretzelBaseCommandParameters(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+        }
+
+        protected override void WithOptions(List<Option> options)
+        {
+            options.AddRange(new[]
+            {
+                new Option(new []{ "-t", "--template" }, "The templating engine to use")
+                {
+                    Argument = new Argument<string>()
+                },
+                new Option(new [] { "-d", "--destination" }, "The path to the destination site (default _site)")
+                {
+                    Argument = new Argument<string>(() => "_site")
+                },
+                new Option("--drafts", "Add the posts in the drafts folder")
+                {
+                    Argument = new Argument<bool>()
+                },
+            });
+        }
+
+        // Default Option that get's injected from Program
+        public string Source { get; set; }
+        // Default Option that get's injected from Program
+        public bool Debug { get; set; }
+        // Default Option that get's injected from Program
+        public bool Safe { get; set; }
+        [Obsolete("Use '" + nameof(Source) + "' instead.")]
```

Comment:
Since we are breaking everything in the next version I think the property can be removed entirely, if it is documented.